### PR TITLE
feat(dlp): added a sample for inspect string custom omit overlap

### PR DESCRIPTION
**Pull request by AarshD-crest**
- Added a sample for [inspect string omit overlap](https://cloud.google.com/dlp/docs/samples/dlp-inspect-string-custom-omit-overlap) with unit test-case.

### CODE REVIEW

File: dlp/snippets/inspect/inspect_string_custom_omit_overlap.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the type and content to be inspected.
+	var contentItem = &dlppb.ContentItem{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_custom_omit_overlap.go:48]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_custom_omit_overlap.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+		DataItem: &dlppb.ContentItem_ByteItem{
+			ByteItem: &dlppb.ByteContentItem{
+				Type: dlppb.ByteContentItem_TEXT_UTF8,
+				Data: []byte(textToInspect),
+			},
+		},
+	}
+	// Construct the custom infotype.
+	var customInfotype = &dlppb.CustomInfoType{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_custom_omit_overlap.go:57]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_custom_omit_overlap.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		Type: &dlppb.CustomInfoType_Regex_{
+			Regex: &dlppb.CustomInfoType_Regex{
+				Pattern: regex,
+			},
+		},
+		ExclusionType: dlppb.CustomInfoType_EXCLUSION_TYPE_EXCLUDE,
+	}
+
+	// Exclude matches that also match the custom infoType.
+	var exclusionRule = &dlppb.ExclusionRule{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_custom_omit_overlap.go:70]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_custom_omit_overlap.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+				InfoTypes: []*dlppb.InfoType{
+					customInfotype.GetInfoType(),
+				},
+			},
+		},
+		MatchingType: dlppb.MatchingType_MATCHING_TYPE_FULL_MATCH,
+	}
+
+	// Construct a ruleset that applies the exclusion rule to the PERSON_NAME infoType.
+	var ruleSet = &dlppb.InspectionRuleSet{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_custom_omit_overlap.go:82]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_custom_omit_overlap.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+			{
+				Type: &dlppb.InspectionRule_ExclusionRule{
+					ExclusionRule: exclusionRule,
+				},
+			},
+		},
+	}
+
+	// Construct the configuration for the Inspect request, including the ruleSet.
+	var config = &dlppb.InspectConfig{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_custom_omit_overlap.go:96]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_custom_omit_overlap.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+	req := &dlppb.InspectContentRequest{
+		Parent:        fmt.Sprintf("projects/%s/locations/global", projectID),
+		Item:          contentItem,
+		InspectConfig: config,
+	}
+
+	// Send the request.
+	resp, err := client.InspectContent(ctx, req)
+	if err != nil {
+		fmt.Fprintf(w, "Receive: %v", err)
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_custom_omit_overlap.go:119]: issue: just return the error to caller.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: dlp/snippets/inspect/inspect_string_custom_omit_overlap.go (hunk 6 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_custom_omit_overlap.go]: Hi @telpirion 
Thank you for all the reviews, 
Fixed the issue over here.
Just want to confirm that, I was just following the [sample style guide error handling guidelines](https://googlecloudplatform.github.io/samples-style-guide/#errors), Could you please resolve the confusion over here?

File: dlp/snippets/inspect/inspect_string_custom_omit_overlap.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_custom_omit_overlap.go]: fixed it.

File: dlp/snippets/inspect/inspect_string_custom_omit_overlap.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_custom_omit_overlap.go]: fixed it.

File: dlp/snippets/inspect/inspect_string_custom_omit_overlap.go (hunk 5 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_custom_omit_overlap.go]: fixed it.

**Review by annadata — APPROVED** [review]
LGTM

File: dlp/snippets/inspect/inspect_string_custom_omit_overlap.go (hunk 6 above)
Comment (telpirion) [reply dlp/snippets/inspect/inspect_string_custom_omit_overlap.go]: Oh, interesting. I wasn't aware that the Sample Style Guide had that guidance. In that case, it's probably okay even if it isn't terribly Go-idiomatic.

**Review by telpirion — APPROVED** [review]

**Pull request merged**